pipeline/data/importers/mono/hplt.py

```diff
@@ -0,0 +1,219 @@
+     - min_fluency_threshold: The minimum score a sentence must have to be included in the final dataset.
+     - max_lines: The maximum number of lines to include in the final dataset.
+     - max_words_in_sentence: The maximum number of words allowed in each sentence.
+     - file_destination: The destination path where the final dataset will be written.
+    """
+
+    with ExitStack() as stack:
+        stats = FilteringStatistics(file_destination)
+
+        outfile = stack.enter_context(write_lines(file_destination))
+
+        shard_urls = load_shard_urls(language)
+        stats.shards.filtered = len(shard_urls)
+        document_stream = stack.enter_context(
+            read_lines(shard_urls, on_enter_location=stats.count_shards_visited)
+        )
+
+        visited_lines = 0
+
+        # Subtract 1 as the final newline is written outside of the for loop.
+        last_line = max_lines - 1
+
+        for document_json in document_stream:
+            stats.document_count.value += 1
+
+            document = HPLTDocument(**json.loads(document_json))
+
+            cumulative_word_count = 0
+            has_written_once = False
+
+            # Visit the lines in the document.
+            for score, lang_item, line in zip(document.scores, document.langs, document.lines):
+                visited_lines += 1
+
+                # Check for the fluency scores.
+                if lang_item == language and score >= min_fluency_threshold:
+                    # TODO(CJK) - Issue #424
+                    word_count = len(line.split())
+
+                    if word_count > max_words_in_sentence:
+                        # This sentence is too long.
+                        cumulative_word_count = 0
+                    else:
+                        stats.visited_lines.kept += 1
+
+                        if has_written_once:
+                            # Determine if this sentence should be added to the previous one or
+                            # written out as a new line. Only concurrent sentences that meet
+                            # the fluency requirement will be combined together.
+                            if (
+                                cumulative_word_count
+                                and cumulative_word_count + word_count < max_words_in_sentence
+                            ):
+                                # Combine sentences together in the outfile.
+                                cumulative_word_count += word_count
+                                outfile.write(" ")
+                            else:
+                                cumulative_word_count = 0
+                                outfile.write("\n")
+                                stats.final_lines.value += 1
+
+                        # Actually write out the line. The next iteration will determine
+                        # if it's a complete sentence, or if it should be added onto.
+                        outfile.write(line)
```

Comment:
where do you do deduplication? It should be done separately for HPLT to make sure we're getting the right number of lines. See https://github.com/mozilla/firefox-translations-training/blob/98f8f1cff25c4a987ea43d8af3a83a8814a4ec84/utils/download_hplt.py#L61

Reply:
Ah, I missed that.